jenkinsci/jira-steps-plugin
# JENKINS-57182 ensure that sites are not added twice through JCASC plugin

See [JENKINS-57182](https://issues.jenkins-ci.org/browse/JENKINS-57182).

## Description

I made 2 commits:
  - The first one ([4a2463b](https://github.com/jenkinsci/jira-steps-plugin/pull/88/commits/4a2463bc619d7cce3d518cb53297315459f2e518)) is just for making it work properly with JCASC plugin.
  - The second ([25edc9c](https://github.com/jenkinsci/jira-steps-plugin/pull/88/commits/25edc9ceb9623e9b4f9cfad423571754153b6c99)) is aesthetic, the JCASC plugin is taking the `Config.java` class name to identify it on the yaml configuration, and there was nothing to identify that chunk of config owns to this plugin's settings. So I've renamed the class and its references from `Config` to `JiraStepsConfig`.

That is why I splitted up the PR into 2 commits.

## Test cases for this fix

There were no tests for this case previously.

## Test plan
  - Run the script snippet on a Jenkins' Script Console and verify that only two sites are added.
  - Run a Jenkins instance with Configuration as Code Plugin installed and the yaml snippet. Verify that only two sites are added.
  - (after 2nd) Restart Jenkins and verify that only 2 sites are configured.

# Submitter checklist

# Reviewer checklist
- [ ] If applicable, tested by installing this plugin on the Jenkins instance.
